flink-dstl/flink-dstl-dfs/src/main/java/org/apache/flink/changelog/fs/RetryingExecutor.java

```diff
@@ -81,11 +111,20 @@ public void close() throws Exception {
      *
      * <p>NOTE: the action must be idempotent because of potential concurrent attempts.
      */
-    interface RetriableAction extends RunnableWithException {}
+    interface RetriableAction<Result> {
+        Result tryExecute() throws Exception;
+
+        void completeWithResult(Result result);
+
+        void discardResult(Result result) throws Exception;
+
+        void handleFailure(Throwable throwable);
```

Comment:
I think it would be clear to add javadocs for these methods. In other words, under what case, will `completeWithResult` be called, and under what case will `discardResult` be called. Moreover, what kind of failure would `handleFailure` handle.